r/src/RcppExports.cpp

```diff
@@ -1281,6 +1344,29 @@ static const R_CallMethodDef CallEntries[] = {
     {"_arrow_read_table_", (DL_FUNC) &_arrow_read_table_, 1},
     {"_arrow_Table__to_dataframe", (DL_FUNC) &_arrow_Table__to_dataframe, 1},
     {"_arrow_Table__column", (DL_FUNC) &_arrow_Table__column, 2},
+    {"_arrow_Array__from_vector", (DL_FUNC) &_arrow_Array__from_vector, 1},
```

Comment:
Non-alphabetical again...

Reply:
No idea what is causing it. I’m using dev version of Rcpp and devtools.

Reply:
@jjallaire @jimhester any ideas on this? More non-deterministic output

Reply:
Sometimes i do build and reload, sometimes load all ... maybe they do it differently but i would think they both call compileAttributes

Reply:
There are 2 passes that determine the registration routines: the first is just referencing our internal list of `// [[Rcpp::export]]` functions, the second are any native routines discovered over and above that by `tools::package_native_routine_registration_skeleton()` (i.e. ones that you don't use `[[Rcpp::export]]` for). In that case perhaps the locale makes a difference and perhaps Build and Reload and devtools::load_all() each do something different w/ the locale?

Reply:
Another thought, the order of these functions seems to be controlled by the order of scanning of source files, which is in turn controlled by the order of files returned from `list.files()`. Perhaps a difference in locale there? (sometimes devtools sets the locale to `C` for determinism/uniformity). 

If the functions within source files were re-ordered that too could affect this.

Reply:
The `Test Package` shortcut seems to disagree with the `Install and Restart`.

Reply:
@kevinushey Is this is a difference in the way RStudio invokes these commands or perhaps a devtools vs. RStudio issue?

Reply:
This is where we list C++ files in Rcpp:

https://github.com/RcppCore/Rcpp/blob/master/R/Attributes.R#L425

We discussed sorting these in the C locale (https://github.com/RcppCore/Rcpp/pull/878) so that we wouldn't introduce these kinds of noisy diffs, but that never made its way to a PR. We should probably make that change.

That said, I don't know why `Test Package` and `Install and Restart` would differ.

Reply:
In the meantime I've set `LC_COLLATE` to `C` in my `~/.Renviron` but that'd be more robust if it was set locally as you hint in 
https://github.com/RcppCore/Rcpp/pull/878

Reply:
I have open https://github.com/RcppCore/Rcpp/issues/911 for following up.